### PR TITLE
projects: fmcdaq2: initial version

### PR DESCRIPTION
**Pull request by amiclaus**
Update FMCDAQ2 project to use latest platform drivers and AXI core implementations.
Folder structure is kept similar to the updated projects.

Add makefile support for FMCDAQ2 project (Makefile, src.mk).

**NOTE**: Current implementation supports the following carrier boards: KCU105, VC707, KC705

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: projects/fmcdaq2/src/devices/adi_hal/parameters.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+#else
+#include "xparameters.h"
+#endif
+
+#ifndef ALTERA_PLATFORM
+#ifdef PLATFORM_MB
+#define SPI_DEVICE_ID				XPAR_SPI_0_DEVICE_ID
+#define GPIO_DEVICE_ID				XPAR_GPIO_0_DEVICE_ID
+
+#define ADC_DDR_BASEADDR			XPAR_AXI_DDR_CNTRL_BASEADDR + 0x800000
```
[the remark this answers is not in the document]
Comment (amiclaus) [reply projects/fmcdaq2/src/devices/adi_hal/parameters.h]: you're right.

File: projects/fmcdaq2/src/app/main.c [hunk 2]
```diff
@@ -0,0 +1,818 @@
...
+int main(void)
+{
+	int32_t status;
+
+	/* Initialize SPI structures */
+	struct spi_init_param ad9523_spi_param = {
+		.max_speed_hz = 2000000u,
+		.chip_select = 0,
+		.mode = SPI_MODE_0
+	};
```
Comment (scuciurean) [inline projects/fmcdaq2/src/app/main.c]: Instead of initializing multiple structures and pass them as a reference, you can declare them inside the main structure.
For example:

```
struct spi_init_param ad9523_spi_param = {
		.max_speed_hz = 2000000u,
		.chip_select = 0,
		.mode = SPI_MODE_0
		.extra = &(struct xil_spi_param) {
				.type = SPI_PL,
				.device_id = SPI_DEVICE_ID
		}
	};
```

This applies for all the structures.

**Comment by amiclaus** [comment]
V2:
- fix indentation in `parameters.h`
- add brackets where needed in the macro definition file

**Comment by dbogdan** [comment]
It's hard to follow the changes. In the first commit I would only move all the files from /fmcdaq2 to projects/fmcdaq2.

**Comment by amiclaus** [comment]
Create a clean PR with requested changes.
